### PR TITLE
Move events out of TermControl.h ;  Use TYPED_EVENT in more places

### PR DESCRIPTION
**Pull request by zadjii-msft**
This is a small refactor on my way to much bigger, more painful refactors. This PR does five things:

* `TermControl.*` has historically had all the control-relevant EventArgs defined in the same file as TermControl. That's just added clutter to the files, clutter that could have been in it's own place.  We'll move all those event arg to their own files. 
* We'll also move `IDirectKeyListener` to its own file while we're at it.
* We'll update some of `TermControl`'s old `DEFINE`/`DECLARE_TYPED_EVENT` macros to the newer `TYPED_EVENT` macro, which is a bit nicer. 
* We'll change `TermControl.TitleChanged` to a typed event. I needed that for a future PR, so let's just do it here
* While we're updating `TYPED_EVENT` macros, let's do `TerminalPage` too.  

### checklist 
* [x] I work here
* [x] This is work for #1256, but we've got a long way to go before that works.

### CODE REVIEW

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 1]
```diff
@@ -3392,14 +3393,6 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // -------------------------------- WinRT Events ---------------------------------
     // Winrt events need a method for adding a callback to the event and removing the callback.
     // These macros will define them both for you.
-    DEFINE_EVENT(TermControl, TitleChanged, _titleChangedHandlers, Control::TitleChangedEventArgs);
     DEFINE_EVENT(TermControl, FontSizeChanged, _fontSizeChangedHandlers, Control::FontSizeChangedEventArgs);
     DEFINE_EVENT(TermControl, ScrollPositionChanged, _scrollPositionChangedHandlers, Control::ScrollPositionChangedEventArgs);
```
Comment (carlos-zamora) [inline src/cascadia/TerminalControl/TermControl.cpp:3397]: Why can't we get rid of these just like we did with the others?

File: src/cascadia/TerminalControl/EventArgs.h [hunk 2]
```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
```
Comment (carlos-zamora) [inline src/cascadia/TerminalControl/EventArgs.h:2]: nit: add an abstract and slap your name on it haha

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 3]
```diff
@@ -3340,16 +3340,4 @@ namespace winrt::TerminalApp::implementation
             }
         }
     }
-
-    // -------------------------------- WinRT Events ---------------------------------
-    // Winrt events need a method for adding a callback to the event and removing the callback.
-    // These macros will define them both for you.
-    DEFINE_EVENT_WITH_TYPED_EVENT_HANDLER(TerminalPage, TitleChanged, _titleChangeHandlers, winrt::Windows::Foundation::IInspectable, winrt::hstring);
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.cpp:3347]: Are we able to kill the `DEFINE_EVENT_WITH_TYPED_EVENT_HANDLER` macro now?

**Review by carlos-zamora — APPROVED** [review]
This makes me happy

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 4]
```diff
@@ -2399,7 +2399,8 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     void TermControl::_TerminalTitleChanged(const std::wstring_view& wstr)
     {
-        _titleChangedHandlers(winrt::hstring{ wstr });
+        auto titleArgs = winrt::make_self<TitleChangedEventArgs>(winrt::hstring{ wstr });
+        _TitleChangedHandlers(*this, *titleArgs);
```
Comment (DHowett) [inline src/cascadia/TerminalControl/TermControl.cpp:2403]: something future peeps can do:

```suggestion
        _TitleChangedHandlers(*this, winrt::make<TitleChangedEventArgs>(winrt::hstring{ wstr }));
```

that's the purpose of make (without self)! :smile:

Comment (DHowett) [reply src/cascadia/TerminalControl/TermControl.cpp:2403]: _i am pretty sure_

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 3 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp:3347]: unsure if that's the last one, I can check though

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalControl/TermControl.cpp:3397]: So, those ones got moved in the branch that this actually came from in the first place, so they didn't have a good clean cherry-pick to work with. There's not a _good_ reason other than possibly laziness

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 4 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalControl/TermControl.cpp:2403]: I can check

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 3 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp:3347]: gottem all

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 4 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalControl/TermControl.cpp:2403]: yep you definitely can

**Pull request merged**